portfolio/src/main/webapp/script.js

```diff
@@ -16,120 +16,195 @@
-        commentsArray.forEach((comment) => {
-            commentsListElement.appendChild(createCommentElement(comment));
-        })
-    });
+  fetch('/data').then(response => response.json()).then((commentsArray) => {
+    const commentsListElement = document.getElementById('comments-list');
+    currentComments = commentsArray;
+    commentsArray.forEach((comment) => {
+      commentsListElement.appendChild(createCommentElement(comment));
+    })
+  });
 }
 
+// Creating comment element with delete button 
 function createCommentElement(comment) {
-    const commentElement = document.createElement('li');
-    commentElement.className = 'comment';
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const usernameElement = document.createElement('div');
+  usernameElement.innerText = comment.username;
+  usernameElement.className = 'namediv';
+  const messageElement = document.createElement('div');
+  messageElement.innerText = comment.message;
+  messageElement.className = 'messagediv';
+
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+  let test = prompt("Enter passcode:", "");
+    if(test == "1234"){
+      deleteComment(comment);
+      var index = currentComments.indexOf(comment);
+      currentComments.splice(index,1);
+      commentElement.remove();
+    }
+  });
+
+  commentElement.appendChild(usernameElement);
+  usernameElement.appendChild(deleteButtonElement);
+  commentElement.appendChild(messageElement);
 
-    const usernameElement = document.createElement('span');
-    usernameElement.innerText = comment.username;
-    usernameElement.className = 'namespan';
-    const messageElement = document.createElement('span');
-    messageElement.innerText = comment.message;
-    usernameElement.className = 'messagespan';
+  return commentElement;
+}
 
-    commentElement.appendChild(messageElement);
-    commentElement.appendChild(usernameElement);
+// Deleting a comment from server
+function deleteComment(comment) {
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  fetch('/delete-comment', {method: 'POST', body: params});
+}
+
+function getData() {
+  var xhr = new XMLHttpRequest();                 // Create XMLHttpRequest object
+
+  xhr.onload = function() {                       // When response has loaded
+    // The following conditional check will not work locally - only on a server
+    if(xhr.status === 200) {
+      var text = xhr.responseText;                   // If server status was ok
+      refreshComments(JSON.parse(text)); // Update
+    }
+  };
+
+  xhr.open('GET', '/data', true);        // Prepare the request
+  xhr.send(null);                                 // Send the request
+}
+
+// Refreshes the data each 2 seconds
+setInterval(getData, 1000*2);
```

Comment:
I would suggest to have this duration as a const at the beginning of the file instead of a magic number.